spikeinterface/core/waveform_extractor.py

```diff
@@ -118,6 +118,67 @@ def create(cls, recording, sorting, folder, remove_if_exists=False):
 
         return cls(recording, sorting, folder)
 
+    @classmethod
+    def register_extension(cls, extension_class):
+        """
+        This maintain a list of possible extension that are available.
+        This depend on the imported submodule.
+        
+        For instance:
+        import spikeinterface as si
+        si.WaveformExtractor.extensions == []
+
+        import spikeinterface.toolkit
+        si.WaveformExtractor.extensions == [WaveformPrincipalComponent, ...]
+        
+        """
+        assert issubclass(extension_class, WaveformExtractorExtensionBase)
+        assert all(extension_class.extension_name != ext.extension_name for ext in cls.extensions), 'Extension name already exists'
+        cls.extensions.append(extension_class)
+
+    def get_extension_class(self, extension_name):
+        """
+        Get extension class from name and check if registered.
+        """
+        extensions_dict = {ext.extension_name: ext for ext in self.extensions}
+        assert extension_name in extensions_dict, \
+            'Extension is not registered, please import related module before'
+        ext_class = extensions_dict[extension_name]
+        return ext_class
+
+    def is_extension(self,  extension_name):
+        """
+        Check if the extension exists in the folder.
+        """
+        ext_class = self.get_extension_class(extension_name)
+        ext_folder = self.folder / ext_class.extension_name
+        params_file = ext_folder / 'params.json'
+        return ext_folder.is_dir() and params_file.is_file()
+
+    def load_extension(self, extension_name):
+        """
+        Load an extension from its name.
+        The module of the extension must be loaded and registered.
+        """
+        ext_class = self.get_extension_class(extension_name)
+        return ext_class.load_from_folder(self.folder)
+
+    def get_extensions_class(self):
+        """
+        Browse persistent extension in the folder.
+        Return list of class.
+        Then instances can be loaded with we.load_extension(extension_name)
+        
+        Importante note : extension module need to be loaded (and so registered)
+        before this call. Otherwise extension will be ignored even if the folder
+        exists.
```

Comment:
```suggestion
        Browse persistent extensions in the folder.
        Return a list of classes.
        Then instances can be loaded with we.load_extension(extension_name)
        
        Importante note: extension modules need to be loaded (and so registered)
        before this call, otherwise extensions will be ignored even if the folder
        exists.
```